### DISCUSSION TITLE
Feedback symbol information to original ast tree

### DISCUSSION THREAD

**Discussion by HungMingWu** [MikePopoloski/slang#628] [General]
Hi Mike, I have some idea about AST transformation
Here is an example:
``` cpp
module top();
   sub #(8) sub ();
endmodule

module sub
#(parameter WIDTH = 16)
();
   reg [WIDTH - 1 : 0] in;
endmodule
```
I hope to remove  parameter, and it looks like
``` cpp
module top();
   sub sub ();
endmodule

module sub
();
   reg [8- 1 : 0] in;
endmodule
```
My naive method is
``` cpp
void transformt( std::vector<SharedSyntaxTree>& syntaxTrees);
{
    ok = driver.parseAllSources();
    tranform(driver.syntaxTrees);
    auto compilation = driver.createCompilation();
}
```
The processing is verbose and error-prone,
Another selection is
``` cpp
void recustruct(const slang::ast::RootSymbol&);
{
    ok = driver.parseAllSources();
    auto compilation = driver.createCompilation();
    recustruct(compilation->getRoot());
}
```
Now I have PortSymbol information,  but I cannot get original syntax tree information.
Which method do you suggest? Or have  better solution?
Thanks

**Comment by MikePopoloski** [comment]
I understand what you're trying to do but the snippets you posted don't really have enough information to say what's wrong or where you're getting stuck. The SyntaxRewriter needs to have the compilation to get info about the parameter values, but other than that I'm not sure how else to help you.
